### PR TITLE
pop: init at 0.1.0

### PR DESCRIPTION
**Pull request by caarlos0**
###### Description of changes

added new package: pop

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by maaslalani — APPROVED** [review]

File: pkgs/applications/misc/pop/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib, buildGoModule, installShellFiles, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "pop";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "charmbracelet";
+    repo = pname;
```
Comment (ambroisie) [inline pkgs/applications/misc/pop/default.nix:9]: Use an explicit `repo = "pop"` instead.

File: pkgs/applications/misc/pop/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      --fish <($out/bin/pop completion fish) \
+      --zsh <($out/bin/pop completion zsh)
+  '';
+
+  meta = with lib; {
+    description = "Send emails from your terminal";
+    homepage = "https://github.com/charmbracelet/pop";
+    changelog = "https://github.com/charmbracelet/pop/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ caarlos0 maaslalani ];
```
Comment (ambroisie) [inline pkgs/applications/misc/pop/default.nix:38]: @maaslalani do you want to be added to the list?

**Review by ambroisie — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/pop/default.nix (hunk 2 above)
Comment (ambroisie) [reply pkgs/applications/misc/pop/default.nix:38]: I hadn't noticed you'd already approved the change, sorry about the ping.

Comment (maaslalani) [reply pkgs/applications/misc/pop/default.nix:38]: Not an issue at all! Thank you

File: pkgs/applications/misc/pop/default.nix (hunk 1 above)
Comment (maaslalani) [inline pkgs/applications/misc/pop/default.nix:9]:
```suggestion
    repo = "pop";
```

**Review by maaslalani — APPROVED** [review]

**Review by ambroisie — APPROVED** [review]

**Review by ambroisie — COMMENTED** [review]
I think the commits should be squashed (I hadn't noticed on my first review).

**Comment by caarlos0** [comment]
@ambroisie sorry, done!

**Review by ambroisie — APPROVED** [review]

**Review by maaslalani — APPROVED** [review]

**Comment by maaslalani** [comment]
@ambroisie Is this good to merge?

**Comment by ambroisie** [comment]
It is, but I don't have merge bits.

**Pull request merged**